tests/test_vertex_executor.py

```diff
@@ -61,9 +61,11 @@ def test_run(self, stage_mock, start_mock, status_mock, _):
         with self.assertRaises(VertexCancelledError):
             self.vertex_executor.run()
 
-        assert stage_mock.called_once()
```

Comment:
These were passing because they don't do anything; `called_once()` returns the mock (with that mock function). It'd be the equivalent to writing: `assert stage_mock.asdfguealsdfka()`.

This does fail in python 3.12, which is nice. But unfortunately one of the mocks (`stage_mock`) isn't working. I confirmed that the issue is with the mock and not with the code. `_stage()` _is_ being called.

Reply:
Nice fix!